lib/elixir/lib/version.ex

```diff
@@ -671,7 +669,7 @@ end
 
 defimpl Inspect, for: Version do
   def inspect(self, _opts) do
-    "#Version<" <> to_string(self) <> ">"
+    "~Version<" <> to_string(self) <> ">"
```

Comment:
Wouldn't that be a breaking change? Out of all other it is the only one that could be used in a form of (checking on PIDs or Refs would almost always fail and URI didn't had `Inspect` implemented):

```elixir
iex> Version.parse("1.2.3")
#Version<1.2.3>
```

Reply:
Not sure I follow, could you elaborate?

Do you mean that someone could have written:

```elixir
"#Version" <> string = inspect(version)
```

and now it'd be a breaking change? FWIW we already broke such promise when we introduced `~U`. I personally wouldn't consider it a breaking change as inspect is using for showing stuff, folks shouldn't couple to that representation, but I hear what you're saying.

Reply:
No, I am talking about [opaque types in DocTests](https://hexdocs.pm/ex_unit/ExUnit.DocTest.html#module-opaque-types).

Reply:
Gotcha. Yeah, these doctests would break.

Reply:
Yes, whenever we change inspected representations, tests may break. However, inspected representations are allowed to change, because the primary goal is developer readability and information. I would say this is even a pro of this PR, because we will push people towards a programmatic representation instead of a textual one.

Reply:
Ok, I am just pointing this out. If you are ok with that, then I am ok. I was pointing this just in case if that wasn't intended.